cudax/include/cuda/experimental/__memory_resource/shared_resource.cuh

```diff
@@ -0,0 +1,273 @@
+  //! @brief Equality comparison between two \c shared_resource
+  //! @param __lhs The first \c shared_resource
+  //! @param __rhs The other \c shared_resource
+  //! @return Checks whether the objects refer to resources that compare equal.
+  _CCCL_NODISCARD_FRIEND bool operator==(const shared_resource& __lhs, const shared_resource& __rhs)
+  {
+    if (__lhs.__control_block == __rhs.__control_block)
+    {
+      return true;
+    }
+
+    if (__lhs.__control_block == nullptr || __rhs.__control_block == nullptr)
+    {
+      return false;
+    }
+
+    return __lhs.__control_block->__resource == __rhs.__control_block->__resource;
+  }
+
+  //! @brief Equality comparison between two \c shared_resource
+  //! @param __lhs The first \c shared_resource
+  //! @param __rhs The other \c shared_resource
+  //! @return Checks whether the objects refer to resources that compare unequal.
+  _CCCL_NODISCARD_FRIEND bool operator!=(const shared_resource& __lhs, const shared_resource& __rhs)
+  {
+    return !(__lhs == __rhs);
+  }
+
+  //! @brief Forwards the stateless properties
+  _LIBCUDACXX_TEMPLATE(class _Property)
+  _LIBCUDACXX_REQUIRES((!property_with_value<_Property>) _LIBCUDACXX_AND(has_property<_Resource, _Property>))
+  friend void get_property(const shared_resource&, _Property) noexcept {}
+
+  //! @brief Forwards the stateful properties
+  _LIBCUDACXX_TEMPLATE(class _Property)
+  _LIBCUDACXX_REQUIRES(property_with_value<_Property> _LIBCUDACXX_AND(has_property<_Resource, _Property>))
+  _CCCL_NODISCARD_FRIEND __property_value_t<_Property> get_property(const shared_resource& __self, _Property) noexcept
+  {
+    return get_property(__self.__control_block->__resource, _Property{});
+  }
+
+private:
+  // Use a custom shared_ptr implementation because (a) we don't need to support weak_ptr so we only
+  // need one pointer, not two, and (b) this implementation can work on device also.
+  struct _Control_block
+  {
+    _Resource __resource;
+    _CUDA_VSTD::atomic<int> __ref_count;
+  };
+
+  _Control_block* __control_block;
+};
+
+//! @rst
+//! .. _cudax-memory-resource-make-shared-resource:
+//!
+//! Factory function for `shared_resource` objects
+//! -----------------------------------------------
+//!
+//! ``make_any_resource`` constructs an :ref:`shared_resource <cudax-memory-resource-shared-resource>` object that wraps
+//! a newly constructed instance of the given resource type. The resource type must satisfy the ``cuda::mr::resource``
+//! concept and provide all of the properties specified in the template parameter pack.
+//!
+//! @param __args The arguments used to construct the instance of the resource type.
+//!
+//! @endrst
+template <class _Resource, class... _Args>
+auto make_shared_resource(_Args&&... __args) -> shared_resource<_Resource>
+{
+  static_assert(_CUDA_VMR::resource<_Resource>, "_Resource does not satisfy the cuda::mr::resource concept");
+  return shared_resource<_Resource>{_CUDA_VSTD::forward<_Args>(__args)...};
+}
```

Comment:
How is `make_shared_resource<_Resource>(args...)` any different than `shared_resource<_Resource>(args...)` apart from being five characters longer?

Reply:
I believe it is just symmetric with all the other stuff